src/liballoc/collections/linked_list.rs

```diff
@@ -878,6 +878,55 @@ impl<T> LinkedList<T> {
         unsafe { self.split_off_after_node(split_node, at) }
     }
 
+    /// Removes the element at the given index and returns it.
```

Comment:
Can you note that the complexity here is `O(n)`?

We can also mention that this is going to scan from the right side.